src/context/MDContext/MDContext.js

```diff
@@ -0,0 +1,191 @@
+const os = require('os');
+const fs = require("fs");
+
+module.exports = {
+
+  contextGenerator: function(path){
+    if (path.endsWith('.md')) {
+      const lines = fs.readFileSync(path, 'utf-8').split(os.EOL);
+      let result = {data: [], sections: 0, links: 0, 
+        emphasis:0, lists: 0, images: 0, quotes: 0, code: 0, tables: 0, texts: 0};
+      
+      for (let i = 0; i < lines.length; i++){
+        if (this.sectionsSearch(lines[i], i) !== undefined){
+          result.data.push(this.sectionsSearch(lines[i], i));
+          result.sections += this.sectionsSearch(lines[i], i).quantity;
+        }
+
+        if (this.textSearch(lines[i], i) !== undefined){
+          result.data.push(this.textSearch(lines[i], i));
+          result.texts += this.textSearch(lines[i], i).quantity;
+        }
+
+        if (this.listsSearch(lines[i], i) !== undefined){
+          result.data.push(this.listsSearch(lines[i], i));
+          result.lists += this.listsSearch(lines[i], i).quantity;
+        }
+
+        if (this.orderedListsSearch(lines[i], i) !== undefined){
+          result.data.push(this.orderedListsSearch(lines[i], i));
+          result.lists += this.orderedListsSearch(lines[i], i).quantity;
+        }
+
+        if (this.linkSearch(lines[i], i) !== undefined){
+          result.data.push(this.linkSearch(lines[i], i));
+          result.links += this.linkSearch(lines[i], i).quantity;
+        }
+        
+        if (this.directLinkSearch(lines[i], i) !== undefined){
+          result.data.push(this.directLinkSearch(lines[i], i));
+          result.links += this.directLinkSearch(lines[i], i).quantity;
+        }
+
+        if (this.inLineCodeSearch(lines[i], i) !== undefined){
+          result.data.push(this.inLineCodeSearch(lines[i], i));
+          result.code += this.inLineCodeSearch(lines[i], i).quantity;
+        }
+      }
+      
+      console.log(result);
```

Comment:
Why not return the result? the output should be responsability of the exit method